test/forward.test.cpp

```diff
@@ -750,4 +752,118 @@ TEST_CASE("autodiff::dual tests", "[dual]")
 	    for(auto j = 0; j < 2; ++j)
 		REQUIRE( x(i,j) == approx(y(i,j)) );
     }
+
+    SECTION("test gradient size with respect to few arguments")
+    {
+        // Testing complex function involving sin and cos
+        auto f = [](const VectorXdual& x, dual y, const VectorXdual& z) -> dual
+        {
+            return 0.5 * (( x.array() * x.array() ).sum() + y * y + (z.array() * z.array()).sum());
+        };
+
+        VectorXdual x(3);
+        x << 1.0, 2.0, 3.0;
+        
+        dual y = 2;
+   
+        VectorXdual z(4);
+        z << 1.0, 2.0, 3.0, 4.0;
+
+        VectorXd g = gradient(f, wrtpack(x.tail(2), y, z), at(x, y, z));
+        
+        REQUIRE( g.size() == 7 );
```

Comment:
Could you also please test the gradient values?

Reply:
Not sure that it is necessary, since they calculated in next test case.